ic-agent/src/identity/basic.rs

```diff
@@ -28,15 +30,56 @@ impl BasicIdentity {
         let bytes: Vec<u8> = pem_reader
             .bytes()
             .collect::<Result<Vec<u8>, std::io::Error>>()?;
+        let file = std::str::from_utf8(&bytes).unwrap();
+        let pem_file: PrivateKeyDocument = file.parse().map_err(|_| PemError::PemError)?;
+        let key_info = pem_file.private_key_info();
+
+        // Check OID
+        if key_info.algorithm.oid != "1.3.101.112".parse().unwrap() {
+            return Err(PemError::KeyRejected(format!(
+                "Wrong OID, expected \"1.3.101.112\", got {}",
+                key_info.algorithm.oid,
+            )));
+        }
+
+        // Check that there are no parameters on the key
+        if key_info.algorithm.parameters.is_some() {
+            return Err(PemError::KeyRejected(
+                "Parameters on the key are not allowed".to_string(),
+            ));
+        }
+
+        // Retrieve the secret key and check that we have an uncompressed
+        // point of 32 byte length
+        let sk = key_info.private_key;
+        if sk[0] != 4 || sk[1] != 32 {
+            return Err(PemError::KeyRejected(
+                "Key is not a ECC private key".to_string(),
+            ));
+        }
+
+        // Retrieve the public key, error if it was not provided
+        let pk = match key_info.public_key {
+            Some(pk) => pk,
```

Comment:
Should we also check the length of the public key?

Reply:
IMO we shouldn't require the PK at all :( 
Instead of checking the public key, I will regerate the `PublicKey` from the `SecretKey` and then compare the two and reject, if they don't match.